src/main/state.gd

```diff
@@ -0,0 +1,11 @@
+class_name State
+extends Node
+
+var frames := 0
+
+func enter(host) -> void:
+	pass
+
+
+func update(host) -> String:
```

Comment:
Static type these host fields -- they have to be "Nodes" if nothing else, because of this class's design

Reply:
Added static typing for Node.

Reply:
This is not a good idea, because static typing the host fields prevents implementors from providing their own types. I'll leave them untyped.